EpiAware/src/EpiLatentModels/broadcast/latentmodel.jl

```diff
@@ -0,0 +1,61 @@
+@doc raw"
+The `BroadcastLatentModel` struct represents a latent model that supports broadcasting of latent periods.
+
+## Constructors
+- `BroadcastLatentModel(model::M; period::Int, broadcast_rule::B)`: Constructs a `BroadcastLatentModel` with the given `model`, `period`, and `broadcast_rule`.
+- `BroadcastLatentModel(model::M, period::Int, broadcast_rule::B)`: An alternative constructor that allows the `model`, `period`, and `broadcast_rule` to be specified without keyword arguments.
+
+## Examples
+```julia
+using EpiAware, Turing
+each_model = BroadcastLatentModel(RandomWalk(), 7, RepeatEach())
+gen_each_model = generate_latent(each_model, 10)
+rand(gen_each_model)
+
+block_model = BroadcastLatentModel(RandomWalk(), 3, RepeatBlock())
+gen_block_model = generate_latent(block_model, 10)
+rand(gen_block_model)
+```
+"
+struct BroadcastLatentModel{
+    M <: AbstractLatentModel, P <: Integer, B <: AbstractBroadcastRule} <:
+       AbstractLatentModel
+    "The underlying latent model."
+    model::M
+    "The period of the broadcast."
+    period::P
+    "The broadcast rule to be applied."
+    broadcast_rule::B
+
+    function BroadcastLatentModel(model::M; period::Integer,
+            broadcast_rule::B) where {M <: AbstractLatentModel, B <: AbstractBroadcastRule}
+        BroadcastLatentModel(model, period, broadcast_rule)
+    end
+
+    function BroadcastLatentModel(model::M, period::Integer,
+            broadcast_rule::B) where {M <: AbstractLatentModel, B <: AbstractBroadcastRule}
+        @assert period>0 "period must be greater than 0"
+        new{typeof(model), typeof(period), typeof(broadcast_rule)}(
+            model, period, broadcast_rule)
+    end
+end
+
+@doc raw"
+Generates latent periods using the specified `model` and `n` number of samples.
+
+## Arguments
+- `model::BroadcastLatentModel`: The broadcast latent model.
+- `n::Any`: The number of samples to generate.
+
+## Returns
+- `broadcasted_latent`: The generated broadcasted latent periods.
+- `latent_period_aux...`: Additional auxiliary information about the latent periods.
+
+"
+@model function EpiAwareBase.generate_latent(model::BroadcastLatentModel, n)
```

Comment:
Just to be clear, with `RepeatEach` with (say) `m=7` this generates 7 random variables. E.g. this is for specific day of week effects and has to be blended with other modelling to have day of week cyclic behaviour along with longer trend effects?

Reply:
Yes exactly that would be the idea. I plan to make a new PR with a `CombineLatentModels` struct and method and another one for `ascertainment` (in `EpiObsModels`).